spec/features/merchants/dashboard_spec.rb

```diff
@@ -1,95 +1,186 @@
+    @transaction12 = Transaction.create!(id: 56, invoice_id: @invoice12.id, credit_card_number:12 , result:0)
+    @transaction13 = Transaction.create!(id: 57, invoice_id: @invoice13.id, credit_card_number:13 , result:0)
+    @transaction14 = Transaction.create!(id: 58, invoice_id: @invoice14.id, credit_card_number:14 , result:0)
+    @transaction15 = Transaction.create!(id: 59, invoice_id: @invoice15.id, credit_card_number:15 , result:0)
+    @transaction16 = Transaction.create!(id: 60, invoice_id: @invoice16.id, credit_card_number:16 , result:0)
+    @transaction17 = Transaction.create!(id: 61, invoice_id: @invoice17.id, credit_card_number:17 , result:0)
+    @transaction18 = Transaction.create!(id: 62, invoice_id: @invoice18.id, credit_card_number:18 , result:0)
+  end
+
   describe 'As a merchant, When I visit my merchant dashboard' do
     it 'I see the name of my merchant' do
-
       visit merchant_dashboards_path(Merchant.first)
 
       expect(page).to have_content(Merchant.first.name)
       expect(page).to_not have_content(Merchant.last.name)
     end
 
     it 'I see link to my merchant items index' do
-
       visit merchant_dashboards_path(Merchant.first)
 
-
       expect(page).to have_link("My Items")
     end
 
     it 'I see link to my merchant invoices index' do
-
       visit merchant_dashboards_path(Merchant.first)
 
       expect(page).to have_link("My Invoices")
     end
 
-    #User story 3: Top 5 Customers
-    xit 'I see the names of the top 5 customers who have conducted the largest number of successful transactions with my merchant' do
-      visit merchant_dashboards_path(Merchant.first)
+    describe 'Top 5 Customers' do
 
-      xwithin ("#top_customers") do
-        expected_hash = {}
+      it 'I see the names of the top 5 customers who have conducted the largest number of successful transactions with my merchant' do
+        visit merchant_dashboards_path(45)
 
-        expect(Merchant.first.top_five_customers_by_merchant).to eq(expected_hash)
+        within ("#top_customers") do
+          customer1 = "Porter Whitehall"
+          customer2 = "Donna Petereit"
+          customer3 = "Carla Whipkey"
+          customer4 = "Becka Hendricks"
+          customer5 = "John Doe"
+
+          expect(customer1).to appear_before(customer2)
+          expect(customer2).to appear_before(customer3)
+          expect(customer3).to appear_before(customer4)
+          expect(customer4).to appear_before(customer5)
+        end
       end
-    end
 
-    xit 'Next to each customer name I see the number of successful transactions they have conducted with my merchant' do
-      visit merchant_dashboards_path(Merchant.first)
+      it 'Next to each customer name I see the number of successful transactions they have conducted with my merchant' do
+        visit merchant_dashboards_path(45)
 
-      expected_hash = {}
+        within "#customer_49" do
+          expect(page).to have_content("Porter Whitehall - 5 purchases")
```

Comment:
we should add a few .to_not have statements in these within blocks during refactoring

Reply:
Test data looks great!!!!